ext/DynamicHMCExt.jl

```diff
@@ -0,0 +1,95 @@
+module DynamicHMCExt
+###
+### DynamicHMC backend - https://github.com/tpapp/DynamicHMC.jl
+###
+
+
+if isdefined(Base, :get_extension)
+    import DynamicHMC
+    using Turing
+    using Turing: AbstractMCMC, Random, LogDensityProblems 
+    using Turing.Inference: LogDensityProblemsAD
+else
+    import ..DynamicHMC
+    using ..Turing
+    using ..Turing: AbstractMCMC, Random, LogDensityProblems 
+    using ..Turing.Inference: LogDensityProblemsAD
+end
+
+# Wraps DynamicHMC as an AbstractSampler
+struct DynamicNUTS{S<:DynamicHMC.NUTS} <: AbstractMCMC.AbstractSampler
+    sampler::S
+end
+
+DynamicNUTS() = DynamicNUTS(DynamicHMC.NUTS())
+Turing.externalsampler(spl::DynamicHMC.NUTS) = Turing.externalsampler(DynamicNUTS(spl))
```

Comment:
Do we need to re-wrap in external here?

Reply:
Both you and @devmotion pointed this out. 
I will change it!

Reply:
I think so! See post below